src/plugins/layout.ts

```diff
@@ -115,7 +115,9 @@ export default class Layout implements IPlugin {
     while (this.originalLayouts.length > 0 && getTimestamp(true) < yieldTime) {
       let originalLayout = this.originalLayouts.shift();
       let originalLayoutState = originalLayout.layout;
-      let currentLayoutState = createLayoutState(originalLayout.node, this.shadowDom);
+      let originalNode = originalLayout.node;
+      let currentStyles = originalNode.nodeType === Node.ELEMENT_NODE ? window.getComputedStyle(originalNode as Element) : null;
+      let currentLayoutState = createLayoutState(originalNode, currentStyles, this.shadowDom);
```

Comment:
Isn't this expensive? Do we need it everywhere? Seems a bit high cost for the value.

Reply:
Not necessarily - the operation is cheap and is something we were already doing for majority of computations when calculating if the element supports scrolling or not. I've added code to measure performance.